### ISSUE TITLE
[Bug]: DN will release a DataSyncSevice twice, cause watch channel event hanging

### ISSUE THREAD

**Issue by XuanYang-cn** [milvus-io/milvus#14604]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version:
- Deployment mode(standalone or cluster):
- SDK version(e.g. pymilvus v2.0.0rc2):
- OS(Ubuntu or CentOS): 
- CPU/Memory: 
- GPU: 
- Others:
```

### Current Behavior

as title.

May have caused the flushing timeout problem.

**Comment by XuanYang-cn** [comment]
fixed by #15102

**Comment by XuanYang-cn** [comment]
Fixed
